libbeat/tests/system/test_template.py

```diff
@@ -105,9 +107,279 @@ def test_json_template(self):
         self.wait_until(lambda: self.log_contains("template with name 'bla' loaded"))
         proc.check_kill_and_wait()
 
-        es = Elasticsearch([self.get_elasticsearch_url()])
+        es = self.es_client()
         result = es.transport.perform_request('GET', '/_template/bla')
         assert len(result) == 1
 
     def get_host(self):
         return os.getenv('ES_HOST', 'localhost') + ':' + os.getenv('ES_PORT', '9200')
+
+
+class TestRunTemplate(BaseTest):
+    """
+    Test run cmd with focus on template setup
+    """
+
+    def setUp(self):
+        super(TestRunTemplate, self).setUp()
+        # auto-derived default settings, if nothing else is set
+        self.index_name = self.beat_name + "-9.9.9"
+
+        self.es = self.es_client()
+        self.idxmgmt = IdxMgmt(self.es)
+        self.idxmgmt.clean(self.beat_name)
```

Comment:
I refactored it a bit now. I don't see a way around deleting indices, alias, template and policies as we need to have a clean state to ensure the setup runs properly, but I ensured that no '*' or empty string can be used. WDYT?

Reply:
Didn't notice we already have had it this way. Kind of a bummer as it might affect test parallelization.

Normally for these kind of tests I use a mix of `<name>-<date>-<hash>` as a prefix. And this per test. Teardown cleans resources only on success, or always after dumping state to some files. But this also requires a few more changes to the beats configs I guess. Adding a prefix is already an improvement. Maybe we should create a ticket to track this.

Reply:
follow up issue https://github.com/elastic/beats/issues/12102